### PR TITLE
msg/async: execute on core specified by core_id not its index

### PR DESCRIPTION
**Pull request by tchaikov**
Reported-by: shangfufei <shangfufei@inspur.com>
Signed-off-by: Kefu Chai <kchai@redhat.com>

### CODE REVIEW

File: src/msg/async/dpdk/DPDKStack.cc [hunk 1]
```diff
@@ -250,8 +250,14 @@ void DPDKStack::spawn_worker(unsigned i, std::function<void ()> &&func)
   // if dpdk::eal::init already called by NVMEDevice, we will select 1..n
   // cores
   assert(rte_lcore_count() >= i + 1);
+  unsigned core_id;
+  RTE_LCORE_FOREACH_SLAVE(core_id) {
+    if (i-- == 0) {
+      break;
+    }
+  }
```
Comment (yuyuyu101) [inline src/msg/async/dpdk/DPDKStack.cc]: what's the meaning?

**Comment by shangfufei** [comment]
i think it's no different with #20638

File: src/msg/async/dpdk/DPDKStack.cc (hunk 1 above)
Comment (shangfufei) [reply src/msg/async/dpdk/DPDKStack.cc]: I think the idea of  #20638 would be better

Comment (tchaikov) [reply src/msg/async/dpdk/DPDKStack.cc]: @yuyuyu101 it's to get the core id from the its index. for example, if i is 0, the first non master core id is used.

Comment (tchaikov) [reply src/msg/async/dpdk/DPDKStack.cc]: @shangfufei, i'd be happy to learn the rationale behind your assertion.

**Comment by tchaikov** [comment]
@yuyuyu101 ping?

File: src/msg/async/dpdk/shared_ptr.h [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```
Comment (yuyuyu101) [inline src/msg/async/dpdk/shared_ptr.h]: just verify, is this update from seastar upstream file?

Comment (tchaikov) [reply src/msg/async/dpdk/shared_ptr.h]: no, it's not. see the commit message of https://github.com/ceph/ceph/pull/20659/commits/d11a8248cfd0c8f14cd10750ceec9326e1fdbee5

**Pull request merged**